subprojects/internal-testing/src/main/groovy/org/gradle/testfixtures/internal/NativeServicesTestFixture.java

```diff
@@ -17,10 +17,13 @@
 package org.gradle.testfixtures.internal;
 
 import org.gradle.internal.nativeintegration.services.NativeServices;
+import org.gradle.test.fixtures.file.TestFile;
 
 import java.io.File;
 
 public class NativeServicesTestFixture {
+    // Collect this early, as the process' current directory can change during embedded test execution
+    public static final TestFile TEST_DIR = new TestFile(new File(".").toURI());
```

Comment:
```suggestion
    private static final TestFile TEST_DIR = new TestFile(new File(".").toURI());
```

This should be private.